### DISCUSSION TITLE
Working with Web3Modal and a browser extension wallet

### DISCUSSION THREAD

**Discussion by michaelessiet** [WalletConnect/walletconnect-monorepo#4143] [Developer Support] [answered]
So I'm creating this browser extension wallet but I don't really know what kind of wallet it needs to be. Should it be an injected wallet, web app or desktop? The current setup I have is the one that is outlined in the [web3wallet docs](https://docs.walletconnect.com/web3wallet/wallet-usage).

What I'm struggling with right now is how do I make the wallet popup whenever a wallet connect connection request is received. This might seem like a dumb question but help would be much appreciated

**Comment by glitch-txs** [comment, marked as answer]
extension wallet = injected.

You will need to support EIP-6963, and the popup should happen in response of 'eth_requestAccounts' method, following EIP-9311
